File: golang/patterns/structural/structural.go
package structural

import (
	"fmt"

	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/structural/adapter"
	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/structural/bridge"
	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/structural/composite"
	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/structural/decorator"
	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/structural/facade"
	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/structural/flyweight"
	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/structural/proxy"
)

func Structural() {
	fmt.Println("\nStructural")

	adapter.Adapter()
	bridge.Bridge()
	composite.Composite()
	decorator.Decorator()
	facade.Facade()
	flyweight.Flyweight()
	proxy.Proxy()
}

File: csharp/pattenrs/Structural.cs
namespace Structural
{
    ///
    /// <summary>
    /// Class <c>StructuralPatterns</c> invokes all Structural patterns for testing purposes.
    /// Structural patterns: concerned about structure (e.g class members)
    /// wrappers that mimic underlying class' interface
    /// stress the importance of good API design
    /// </summary>
    ///
    class StructuralPatterns
    {
        public static void Run()
        {
            Console.WriteLine("\nStructural");

            Adapter.Main.Run();
            Bridge.Main.Run();
            Composite.Main.Run();
            Decorator.Main.Run();
            Facade.Main.Run();
            Flyweight.Main.Run();
            Proxy.Main.Run();
        }
    }
}
